lexos/processors/analyze/content_analysis_model.py

```diff
@@ -0,0 +1,461 @@
+                        ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9",
+                         " ", "+", "-", "*", "/", "sin", "cos", "tan", "log",
+                         "sqrt", "(", ")"]
+        for item in allowed_input:
+            formula = formula.replace(item, "")
+        if len(formula) == 0:
+            return True
+        return False
+
+    def save_formula(self):
+        """saves formula from front-end in the class member _formula"""
+        if self._test_options is not None:
+            formula = self._test_options.formula
+        else:
+            formula = self.content_analysis_option.formula
+        if len(formula) == 0:
+            self._formula = "0"
+        else:
+            formula = formula.replace("√", "sqrt").replace("^", "**")
+            self._formula = formula
+        return self.check_formula()
+
+    def check_formula(self) -> str:
+        """Checks if there are any errors in the formula
+
+        :return: error message if there is an empty str or None if there is
+                 no errors
+        """
+        error_msg = "Formula errors:<br>"
+        is_error = False
+        if self._formula.count("(") != self._formula.count(")"):
+            error_msg += "Mismatched parenthesis<br>"
+            is_error = True
+        if "sin()" in self._formula:
+            error_msg += "sin takes exactly one argument (0 given)<br>"
+            is_error = True
+        if "cos()" in self._formula:
+            error_msg += "cos takes exactly one argument (0 given)<br>"
+            is_error = True
+        if "tan()" in self._formula:
+            error_msg += "tan takes exactly one argument (0 given)<br>"
+            is_error = True
+        if "log()" in self._formula:
+            error_msg += "log takes exactly one argument (0 given)<br>"
+            is_error = True
+        if is_error:
+            return error_msg
+        return ""
+
+    def analyze(self) -> [str, str]:
+        """Generates html table containing counts, averages, & formula results
+
+        :return: formula_errors: str with error message or None if there is no
+                 errors.
+                 result_table: str containing result html table or None if
+                 there are any errors in the formula
+        """
+        self.count()
+        if self.is_secure():
+            formula_errors = self.save_formula()
+            self.generate_scores()
+            self.generate_averages()
+            result_table = self.to_html()
+        else:
+            formula_errors = "Formula error: Invalid input"
+            result_table = ""
+        return result_table, formula_errors
+
+    @property
+    def dictionaries(self) -> list:
+        return self._dictionaries
+
+    @dictionaries.setter
+    def dictionaries(self, dictionaries: list):
+        self._dictionaries = dictionaries
+
+    @property
+    def corpus(self) -> list:
+        return self._corpus
+
+    @property
```

Comment:
Most of these properties should probably be in a NamedTuple. If not, you need docstrings

Reply:
@chantisnake do we really need docstrings for getters and setters?

Reply:
you need to have doc string for every functions. Otherwise it will not pass the test we will implement in the future.